parser/src/context.rs

```diff
@@ -0,0 +1,60 @@
+use lexer::token::TokenType;
+
+use crate::ast::operation::{ARITHMETICS, BinaryOperator, BOOLEANS, COMPARISONS};
+
+#[derive(Clone)]
+/// A structure that contains contexts
+pub struct ParserContext {
+    pub allowed_operators: Vec<BinaryOperator>,
+    pub parsing_value: bool,
+    pub enclosing_end: Option<TokenType>,
+}
+
+impl Default for ParserContext {
+    fn default() -> Self {
+        Self {
+            allowed_operators: [BOOLEANS].concat(),
```

Comment:
Having to allocate a new vector each time a new `ParserContext` is a lot.

Reply:
i wanted to make constants and statics but it seems not to be possible :(